### PR TITLE
Remove invalid dates

### PR DESCRIPTION
**Pull request by nathanathan**
This removes invalid date ranges where the end occurs before the start and prevents them from being created. It also changes the measure of date precision to the length of the date interval.

### CODE REVIEW

File: server/updaters.coffee [hunk 1]
```diff
@@ -191,5 +191,12 @@ module.exports = ->
         incident.modifiedDate = new Date()
         Incidents.update(_id: incident._id, incident)
 
+  console.log "Removing incidents with invalid dates..."
+  invalidDateIncidents = []
+  Incidents.find().map (incident) ->
+    if incident.dateRange.end < incident.dateRange.start
+      invalidDateIncidents.push(incident._id)
+  console.log "Removing #{invalidDateIncidents.length} incidents"
+  Incidents.remove(_id: $in: invalidDateIncidents)
```
Comment (jgoley) [inline server/updaters.coffee]: Should we remove the entire incident? What if the incident's other data is accurate?

Comment (nathanathan) [inline server/updaters.coffee]: The incidents are kind of all or none when it comes to the location, count and date info. If they're missing one those those fields we can't use them for anything, so I'm alright with removing the full incident. Also, these are mostly or all auto-generated so we'll could eventually regenerate them after the processing code has been updated.

Comment (jgoley) [inline server/updaters.coffee]: I see. Thanks for the explanation!

**Comment by jgoley** [comment]
👍

**Pull request merged**